### PR TITLE
snap: update build to noble

### PR DESCRIPTION
**Pull request by Meulengracht**
New in noble version of network-manager:

Netplan patches are replaced by `--enable-netplan=yes`. 
Update libraries needed to build network-manager. U
Update switches that is required for network-manager build.
Update primed library list.

### CODE REVIEW

File: snap-patch/networkmanager/0007-core-do-not-check-match-settings-when-comparing-conn.patch
Comment (alfonsosanchezbeato) [inline snap-patch/networkmanager/0007-core-do-not-check-match-settings-when-comparing-conn.patch:1]: What's the reason for dropping this one? Is it already fixed upstream?

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -160,35 +160,44 @@ parts:
...
       - libndp-dev
       - libreadline-dev
       - libnewt-dev
       - libteam-dev
       - libjansson-dev
-      - libnss3-dev
+      - libbluetooth-dev
+      - valac
+      - polkitd
+      - libpolkit-gobject-1-dev
```
Comment (alfonsosanchezbeato) [inline snap/snapcraft.yaml:191]: Is this really needed? We are disabling polkit in the build.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -160,35 +160,44 @@ parts:
...
-      - systemd
+      - systemd-dev
       - libsystemd-dev
       - libudev-dev
       - libgirepository1.0-dev
       - gobject-introspection
       - python3-gi
       - libpsl-dev
       - libcurl4-gnutls-dev
+      - gtk-doc-tools
```
Comment (alfonsosanchezbeato) [inline snap/snapcraft.yaml:180]: We are disabling gtkdoc, is this needed?

**Review by alfonsosanchezbeato — COMMENTED** [review]
Thanks for this. Some comments, also:
* I think we should drop `0003-Do-not-enable-for-APs-SAE-if-key-management-is-psk.patch` too, we can revert if we detect this is needed in the end. Hopefully NM upstream has sorted out already this sort of situation.
* Aren't we using meson to build now the deb?

File: snap-patch/networkmanager/0007-core-do-not-check-match-settings-when-comparing-conn.patch
Comment (Meulengracht) [reply snap-patch/networkmanager/0007-core-do-not-check-match-settings-when-comparing-conn.patch:1]: This is fixed upstream

**Comment by Meulengracht** [comment]
> Thanks for this. Some comments, also:
> 
> * I think we should drop `0003-Do-not-enable-for-APs-SAE-if-key-management-is-psk.patch` too, we can revert if we detect this is needed in the end. Hopefully NM upstream has sorted out already this sort of situation.
> * Aren't we using meson to build now the deb?

1. I'll drop it
2. Not what I can tell, it still looks like it's using autotools (however there are meson build files)

File: snap/snapcraft.yaml (hunk 2 above)
Comment (Meulengracht) [reply snap/snapcraft.yaml:180]: Yep it's still needed for the build
`./autogen.sh: 25: gtkdocize: not found`

File: snap/snapcraft.yaml (hunk 1 above)
Comment (Meulengracht) [reply snap/snapcraft.yaml:191]: Yep, they are still required for the build, without them we get

`/usr/bin/msgfmt: cannot locate ITS rules for data/org.freedesktop.NetworkManager.policy.in`

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: network-manager
-version: 1.36.6-10-dev
+version: 1.46.0-1ubuntu2
```
Comment (alfonsosanchezbeato) [inline snap/snapcraft.yaml]: version should be `1.46.0-1-dev`

**Review by alfonsosanchezbeato — COMMENTED** [review]
One minor thing, but also, we need to change spread.yaml to enable UC24 tests.

**Pull request merged**